docs/coding-guidelines.md

```diff
@@ -0,0 +1,403 @@
+
+## Copyright header and license notice
+
+All source code files (mostly `src/**/*.cs` and `test/**/*.cs`) require this exact header (please do not make any changes to it):
+
+```cs
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+```
+
+It is ok to skip it on generated files, such as `*.designer.cs`.
+
+Every repo also needs the Apache 2.0 License in a file called LICENSE.txt in the root of the repo. 
+
+## C# Coding Style
+
+The NuGet.Client coding style is similar to the [.NET Framework Coding style](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md) with some minor alterations that consider the history of the client repository.
+
+For non code files (xml, etc), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
+
+The general rule we follow is "use Visual Studio defaults".
+
+1. We use [Allman style](http://en.wikipedia.org/wiki/Indent_style#Allman_style) braces, where each brace begins on a new line. One exception is that a `using` statement is permitted to be nested within another `using` statement by starting on the following line at the same indentation level, even if the nested `using` contains a controlled block.
+
+1. We use four spaces of indentation (no tabs).
+
+1. We use `_camelCase` for internal and private fields and use `readonly` where possible. Prefix internal and private instance fields with `_`. Static fields are all CamelCase regardless of visibility. When used on static fields, `readonly` should come after `static` (e.g. `static readonly` not `readonly static`).  Public fields should be used sparingly and should use PascalCasing with no prefix when used.
+
+1. We avoid `this.` unless absolutely necessary.
+
+1. We always specify the visibility, even if it's the default (e.g.
+   `private string _foo` not `string _foo`). Visibility should be the first modifier (e.g.
+   `public abstract` not `abstract public`).
+
+1. Namespace imports should be specified at the top of the file, *outside* of
+   `namespace` declarations, and should be sorted alphabetically, with the exception of `System.*` namespaces, which are to be placed on top of all others.
+
+1. Avoid more than one empty line at any time. For example, do not have two
+   blank lines between members of a type.
+1. Avoid spurious free spaces.
+   For example avoid `if (someVar == 0)...`, where the dots mark the spurious free spaces.
+   Consider enabling "View White Space (Ctrl+R, Ctrl+W)" or "Edit -> Advanced -> View White Space" if using Visual Studio to aid detection.
```

Comment:
When the markdown is rendered into HTML, the new lines get removed. Given earlier list items have multiple sentences on the same line, it's unclear if the intention here is to have one sentence per line, but in the same HTML paragraph, or if the intention was to have each sentence be rendered as a separate HTML paragraph.

Reply:
I have it that way to improve readability of the doc.

Reply:
changed it though for consistency.